Documentation/data-type-and-flow-control.md

```diff
@@ -73,19 +81,35 @@ Dimensions = (string) Dimension
 
-## other/tensors-flexible
+```other/tensors``` is defined to have multiple instances of tensors in a buffer of a single stream path.
+Compared to having multiple streams (thus, multiple pads) with a single tensor that goes or
+
+
+## other/tensors,format=static (default)
 
-```other/tensors-flexible``` handles non-static, flexible tensor stream without specifying the data type and shape of tensor in pad capability.
+```other/tensors,format=static``` is the default mode for ```other/tensors```.
+With static, the properties of ```types``` and ```dimensions``` are effective and the corresponding pads are supposed to have the fixed size of frames.
+To update dimensions and types of a pad, the pipeline needs to re-negotiate the capability, which is usually not trivial.
+
+
+Note that static is supposed to be most efficient and fast format and if you do not have clear needs for the flexibility, it is recommended to use static.
+
+## other/tensors,format=flexible
+
+```other/tensors,format=flexible``` handles non-static, flexible tensor stream without specifying the data type and shape of tensor in pad capability.
 This is useful when an element or a model requires non-determined, dynamic data shape to process the tensors. (e.g., cropping the raw data into multiple tensors)
 
-Unlike ```other/tensor``` and ```other/tensors```, flexible tensor does not contain the data structure in pad capability.
-Instead, flexible tensor has its own data structure - [GstTensorMetaInfo](https://github.com/nnstreamer/nnstreamer/blob/main/gst/nnstreamer/include/tensor_typedef.h) - in each tensor buffer, to prevent caps negotiation with fixed type of data stream.
-When processing a buffer with the capability ```other/tensors-flexible```, developer should append or parse the tensor information in buffer using various [utility functions](https://github.com/nnstreamer/nnstreamer/blob/main/gst/nnstreamer/include/nnstreamer_plugin_api.h).
+Unlike ```other/tensors,format=static```, flexible tensor does not contain the data structure in pad capability; thus, the properites of dimensions and types are ignored.
+An ```other/tensors,format=flexible``` tensor has its meta data - [GstTensorMetaInfo](https://github.com/nnstreamer/nnstreamer/blob/main/gst/nnstreamer/include/tensor_typedef.h) - in each tensor buffer, to prevent caps negotiation with fixed type of data stream.
+When processing a buffer with the capability ```other/tensors,format=flexible```, NNStreamer engine developer (not pipeline or application developers) should append or parse the tensor information in buffer using various [utility functions](https://github.com/nnstreamer/nnstreamer/blob/main/gst/nnstreamer/include/nnstreamer_plugin_api.h).
```

Comment:
Discussion: now util functions to handle tensor-meta are moved to nnstreamer_plugin_api_util.h (except two functions -appending or parsing header of gst-memory).

Reply:
Added